pkg/ebpf/bpf_client.go

```
@@ -422,7 +440,137 @@ func recoverBPFState(eBPFSDKClient goelf.BpfSDKClient, policyEndpointeBPFContext
 		}
 	}
 
-	return isConntrackMapPresent, isPolicyEventsMapPresent, eventsMapFD, nil
+	//If update required, cleanup probes and gather data to re attach prpbes with new programs
+	if updateIngressProbe || updateEgressProbe {
+		// Get all loaded programs and maps
+		bpfState, err := eBPFSDKClient.GetAllBpfProgramsAndMaps()
+		if err != nil {
+			log.Info("GetAllBpfProgramsAndMaps failed: ", "error: ", err)
```

shall we return a error here instead of continue?

---

updated